File: Grind75/01_TwoSum/two_sum.py
#
# https://leetcode.com/problems/two-sum
# 
# This was super fast to write so I should start setting up a time for completion.
#
# Runtime 3860ms (18.84%)
# Memory  17.2MB (32.79%)
#
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        for i in range(len(nums)):
            for j in range(i+1, len(nums)):
                if (nums[i] + nums[j]) == target:
                    return [i, j]

File: Grind75/01_TwoSum/two_sum.rs
/*
 * https://leetcode.com/problems/two-sum/
 * 
 * I was attempting to use the built in vector for loops.
 * 
 * RunTime  30ms (24.78%)
 * Memory  2.1MB (81.47%)
 */

impl Solution {
  pub fn two_sum(nums: Vec<i32>, target: i32) -> Vec<i32> {
      let mut i = 0;
      let mut j = 0;
      while i < nums.len(){
          j = i + 1; 
          while j < nums.len(){
              if((nums[i]+nums[j]) == target){
                  return vec![i as i32, j as i32];
              }
              j+=1;
          }
          i+=1;
      }
      return vec![0,0];
  }
}